nipoppy/workflows/dataset_init.py

```diff
@@ -85,13 +107,65 @@ def run_main(self):
             FPATH_SAMPLE_MANIFEST, self.layout.fpath_manifest, log_level=logging.DEBUG
         )
 
+        if self.bids_source is not None:
+            self._init_manifest_from_bids_dataset()
+
         # inform user to edit the sample files
         self.logger.warning(
             f"Sample config and manifest files copied to {self.layout.fpath_config}"
             f" and {self.layout.fpath_manifest} respectively. They should be edited"
             " to match your dataset"
         )
 
+    def _init_manifest_from_bids_dataset(self) -> None:
+        """Assume a BIDS dataset with session level folders.
+
+        No BIDS validation is done.
+        """
+        df = {"participant_id": [], "visit_id": [], "session_id": [], "datatype": []}
+        participant_ids = sorted(
+            [
+                x.name
+                for x in (self.layout.dpath_bids).iterdir()
+                if x.is_dir() and x.name.startswith("sub-")
+            ]
+        )
+
+        self.logger.info("Creating a manifest.csv from the BIDS dataset content.")
+
+        for ppt in participant_ids:
+
+            session_ids = sorted(
+                [
+                    x.name
+                    for x in (self.layout.dpath_bids / ppt).iterdir()
+                    if x.is_dir() and x.name.startswith("ses-")
+                ]
+            )
+            if not session_ids:
+                self.logger.warning(
+                    f"Skipping subject '{ppt}': could not find a session level folder."
+                )
+                continue
+
+            for ses in session_ids:
+                datatypes = sorted(
+                    [
+                        x.name
+                        for x in (self.layout.dpath_bids / ppt / ses).iterdir()
+                        if x.is_dir()
+                    ]
+                )
+
+                df["participant_id"].append(ppt.replace("sub-", ""))
+                df["session_id"].append(ses.replace("ses-", ""))
+                df["datatype"].append("[" + "'" + "' ,'".join(datatypes) + "'" + "]")
+
+        df["visit_id"] = df["session_id"]
```

Comment:
```suggestion
                df[Manifest.col_participant_id].append(check_participant_id(ppt))
                df[Manifest.col_session_id].append(check_session_id(ses))
                df[Manifest.col_datatype].append(datatypes)

        df[Manifest.col_visit_id] = df[Manifest.col_session_id]
```

Not 100% but I don't think `datatypes` needs to be converted to a string like that